athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/substrait/SubstraitFunctionParser.java

```diff
@@ -201,38 +280,29 @@ private static boolean isLogicalOperator(String functionName)
 
     /**
      * Maps Substrait function names to corresponding Operator enum values.
-     * This method is mapping only small set of operators, and we will extend this as we need.
+     * This method supports comparison operators, logical operators, null checks, and the NOT operator.
+     * The mapping will be extended as additional operators are needed.
      * 
-     * @param functionName The Substrait function name (e.g., "gt:any_any", "equal:any_any")
+     * @param functionName The Substrait function name (e.g., "gt:any_any", "equal:any_any", "not:bool")
      * @return The corresponding Operator enum value
      * @throws UnsupportedOperationException if the function name is not supported
      */
     private static SubstraitOperator mapToOperator(String functionName)
```

Comment:
I don't think federation supports java17 to use this switch case. Please make sure your all the PR checks and build are successful